### ISSUE TITLE
Error while trying to add new domain

### ISSUE THREAD

**Issue by aylton-almeida** [milliHQ/terraform-aws-next-js-image-optimization#68]
So far, I had the following configuration, and it worked great.

![image](https://user-images.githubusercontent.com/38439416/133479010-541eff0e-9340-4b34-b9cb-27a0606152b9.png)

But when I added the second domain in the list the error bellow started to appear.

![image](https://user-images.githubusercontent.com/38439416/133478801-939b701a-0468-4a9f-a3c7-e0ea70088173.png)

Apparently, it's an error related to the provider. Are you aware of this issue? How can we fix it?

**Comment by ofhouse** [comment]
Hi, the error mostly occurs when we release a new version of the module (Like we did yesterday).
Every new release of the module changes the content of the Lambda (that's handling the optimization work).

Since the content of the Lambda is updated while terraform apply (not terraform init) runs, the content of the Lambda has a different source code hash at the beginning and at the end of the Terraform apply command.
This is basically what the error message is trying to say.

I currently have no permanent fix for this issue, but running `terraform apply` a second time should fix it.

**Comment by ofhouse** [comment]
Related issue in the Terraform AWS provider: https://github.com/hashicorp/terraform-provider-aws/issues/14267

**Comment by ofhouse** [comment]
We could probably workaround the issue by waiting for the download to be finished, before publishing the output in the download module.
Currently the file path is immediately returned from the module and the download happens in parallel.
When we wait with the return of the file path until the download is finished, we should be able to produce the correct result.

Opened a new issue in our download module, will try to include a fix in the next release: https://github.com/milliHQ/terraform-npm-download/issues/3

**Comment by ofhouse** [comment]
Updated the download script so this error should not appear when upgrading to future versions.
Thanks for reporting! 👍
